File: WordCount/wc.py
from __future__ import print_function

import sys
from operator import add

from pyspark.sql import SparkContext
from pyspark.sql import SparkConf


if __name__ == "__main__":
    filein = sys.argv[0]
    fileout = sys.argv[1]

    conf = SparkConf().setMaster("local[*]") \
        .setAppName("WCPythonApp")
    sc = SparkContext(conf)

    text_file = sc.textFile("file://" + filein)
    counts = text_file.flatMap(lambda line: line.split(" ")) \
        .map(lambda word: (word, 1)) \
        .reduceByKey(lambda a, b: a + b)
    counts.saveAsTextFile("file://" + fileout)


spark.stop()

File: WordCount/wc.scala
import org.apache.spark.SparkConf
import org.apache.spark.SparkContext


object WordCouter {
  def main(args: Array[String]): Unit = {
    val filein = args(0)
    val fileout = args(1)
    // create a spark context with desired configs and
    // pass to the sqlcontext
    val conf = new SparkConf().setMaster("local[*]")
      .setAppName("WCScalaApp")
    val sc = new SparkContext(conf)

    val textFile = sc.textFile("file://" + filein)
    val counts = textFile.flatMap(line => line.split(" "))
                     .map(word => (word, 1))
                     .reduceByKey(_ + _)
    counts.saveAsTextFile("file://" + fileout)
    sc.stop()
  }
}
